File: brackets-extended-edition.js
let regex = /[\(\)]\w*[\(\)]|[{}]\w*[{}]|[\[\]]\w*[\[\]]|[<>]\w*[<>]/g;
let replace = s => regex.test(s) ? replace(s.replace(regex, '')) : s;
[...Array(+readline())].map(_ => console.log(!/[\[\]\(\){}<>]/.test(replace(readline()))));

File: brackets-extended-edition.cs
using System;
using System.Text.RegularExpressions;
class S
{
    static void Main()
    {
        string regex = @"[\(\)]\w*[\(\)]|[{}]\w*[{}]|[\[\]]\w*[\[\]]|[<>]\w*[<>]";
        for (int i = 0, N = int.Parse(Console.ReadLine()); i < N; i++)
        {
            string text = Console.ReadLine();
            while(Regex.Match(text, regex).Success) text = Regex.Replace(text, regex, "");
            Console.WriteLine((!Regex.Match(text, @"[\[\]\(\){}<>]").Success+"").ToLower());
        }
    }
}
